pandas/core/frame.py

```diff
@@ -4635,7 +4636,16 @@ def quantile(self, q=0.5, axis=0, numeric_only=True):
             0 <= q <= 1, the quantile(s) to compute
         axis : {0, 1, 'index', 'columns'} (default 0)
             0 or 'index' for row-wise, 1 or 'columns' for column-wise
```

Comment:
@jreback 
The signature of the method for `Series` and `DataFrame` is different. `DataFrame.quantile` has `axis` and `numeric_only` parameters that `Series.quantile` doesn't have. Moreover, the `Returns` and `Examples` are different. How would we setup `_shared_docs` for these?
Do you means separate `_shared_docs` for `Dataframe.quantile` and `Series.quantile` like `_shared_docs['dfquantile]` and `_shared_docs['serquantile']` or a common one with `@Substitution` to replace the respective parameters, Returns and Examples ?

Also I don't understand where `NotImplemented` would go.